### PR TITLE
[WIP] Renode CI

### PR DESCRIPTION
**Pull request by mateusz-holenko**
**DO NOT MERGE IT**

This is a working PR to test Renode tests in CI.

### CODE REVIEW

File: scripts/test-renode.sh [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	# Download prebuilt renode Release if none is currently installed
+	conda install -c antmicro -c conda-forge renode
+	RENODE_BIN=$CONDA_PREFIX/bin/renode
+fi
+
+case $CPU in
+	vexriscv | picorv32)
+		;;
+	*)
+        echo "Simulating the current configuration is not supported in Renode yet - skipping the test"
```
Comment (mithro) [inline scripts/test-renode.sh]: Indenting here looks a bit weird.

File: scripts/test-renode.sh [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	--resc "$RENODE_RESC" \
+	--bios-binary "$TOP_DIR/$TARGET_BUILD_DIR/software/bios/bios.bin" \
+	--firmware-binary "$TOP_DIR/$TARGET_BUILD_DIR/software/$FIRMWARE/firmware.bin"
+
+# TODO: fix this - do not use hardcoded conda-based renode
+OPT_RENODE_LOCATION=$TOP_DIR/build/conda/opt/renode
+
+cd $OPT_RENODE_LOCATION
+
+# TODO: those files should be shipped together with Renode
```
Comment (mithro) [inline scripts/test-renode.sh]: Can you add why these are needed?

File: scripts/test-renode.sh (hunk 1 above)
Comment (mateusz-holenko) [reply scripts/test-renode.sh]: It was caused by mixed tabs & spaces. Fixed now.

File: scripts/test-renode.sh (hunk 2 above)
Comment (mateusz-holenko) [reply scripts/test-renode.sh]: They are needed to run robot tests. The fact they are not in the conda package is a bug we need to fix.

wgetting the missing files is just a workaround to verify that everything else works as expected

**Comment by mateusz-holenko** [comment]
The current version is working. Closing this one and opening a new one to `litex-buildenv/master`.

**Pull request closed without merging**